Foundation/NSCache.swift

```diff
@@ -20,15 +20,49 @@ private class NSCacheEntry<KeyType : AnyObject, ObjectType : AnyObject> {
     }
 }
 
+fileprivate class NSCacheKey: NSObject {
+    
+    var value: AnyObject
+    
+    init(_ value: AnyObject) {
+        self.value = value
+        super.init()
+    }
+    
+    override var hashValue: Int {
+        switch self.value {
+        case let nsObject as NSObject:
+            return nsObject.hashValue
+        case let hashable as Hashable:
+            return hashable.hashValue
+        default: return 0
+        }
+    }
+    
+    override func isEqual(_ object: Any?) -> Bool {
+        guard let other = (object as? NSCacheKey) else { return false }
+        
+        if self.value === other.value {
+            return true
+        } else {
+            guard let left = self.value as? NSObject,
+                let right = other.value as? NSObject else { return false }
+            
+            return left.isEqual(right)
+        }
+    }
+}
+
 open class NSCache<KeyType : AnyObject, ObjectType : AnyObject> : NSObject {
```

Comment:
I dont know if we could really pull this off but I think the "right" fix here would be to change the KeyType to just Hashable since the Any bridging allows all structural types to pass to objc, and the ObjectType should be Any as well. (But that would require compiler changes and overlay fixes for the Darwin version)